kani-driver/src/exe_trace.rs

```diff
@@ -0,0 +1,288 @@
+            args.push("--unstable-features".into());
+            args.push("--file-lines".into());
+            let line_range_dicts_combined = format!("[{}]", line_range_dicts.join(","));
+            args.push(line_range_dicts_combined.into());
+        }
+
+        for file_line_range in file_line_ranges {
+            args.push((&file_line_range.file).into());
+        }
+
+        cmd.args(args);
+
+        if let Some(current_dir) = current_dir_opt {
+            cmd.current_dir(current_dir);
+        }
+
+        if self.args.quiet {
+            self.run_suppress(cmd).expect("Couldn't rustfmt source file");
+        } else {
+            self.run_terminal(cmd).expect("Couldn't rustfmt source file");
+        }
+    }
+}
+
+/// Generate a unit test from a list of det vals.
+fn format_unit_test(
+    harness_name: &str,
+    det_vals: &[Vec<u8>],
+    interp_det_vals: &[String],
+) -> (String, String) {
+    /*
+    Given a number of byte vectors, format them as:
+    // interp_det_val_1
+    vec![det_val_1],
+    // interp_det_val_2
+    vec![det_val_2], ...
+    */
+    let vecs_as_str = det_vals
+        .iter()
+        .zip(interp_det_vals.iter())
+        .map(|(det_val, interp_det_val)| {
+            format!("        // {interp_det_val}\n        vec!{:?}", det_val)
+        })
+        .collect::<Vec<String>>()
+        .join(",\n");
+
+    // Hash the generated det val string along with the proof harness name.
+    let mut hasher = DefaultHasher::new();
+    harness_name.hash(&mut hasher);
+    vecs_as_str.hash(&mut hasher);
+    let hash = hasher.finish();
+
+    let exe_trace_func_name = format!("kani_exe_trace_{harness_name}_{hash}");
+    #[rustfmt::skip]
+    let exe_trace = format!(
+"#[test]
+fn {exe_trace_func_name}() {{
+    let det_vals: Vec<Vec<u8>> = vec![
+{vecs_as_str}
+    ];
+    kani::exe_trace_init(det_vals);
+    {harness_name}();
+}}"
+    );
+
+    (exe_trace, exe_trace_func_name)
+}
+struct FileLineRange {
+    file: String,
+    line_range: Option<(usize, usize)>,
+}
+
+/// Read the CBMC output, parse it as a JSON object, and extract the deterministic values.
+mod parser {
+    use crate::util::append_path;
+    use serde_json::Value;
+    use std::fs::File;
+    use std::io::BufReader;
+    use std::path::Path;
+
+    /// Extract deterministic values from a failing harness.
+    pub fn get_det_vals(file: &Path) -> (Vec<Vec<u8>>, Vec<String>) {
+        let output_filename = append_path(file, "cbmc_output");
+        let cbmc_out = get_cbmc_out(&output_filename);
+        handle_cbmc_out(&cbmc_out)
+    }
+
+    /// Read in the CBMC results file and deserialize it to a JSON object.
+    fn get_cbmc_out(results_filename: &Path) -> Value {
+        let results_file = File::open(results_filename).unwrap();
+        let reader = BufReader::new(results_file);
+        serde_json::from_reader(reader).unwrap()
+    }
+
+    /// The first-level JSON object parser. This extracts the result message.
```

Comment:
Can you expand this comment to say what are you expecting from CBMC output? You can even paste a dummy example on how this looks.

Reply:
Done